Code/src/main/java/nl/utwente/group10/ui/components/blocks/FunctionBlock.java

```diff
@@ -327,4 +352,20 @@ public final void error() {
     public OutputAnchor getOutputAnchor() {
         return output;
     }
+
+    @Override
+    public void changed(ObservableValue<? extends Number> arg0, Number arg1, Number arg2) {
+        if (arg0.equals(bowtieIndex)) {
+            invalidateBowtieIndex();
+        }
+    }
+
+    private void invalidateBowtieIndex() {
+        for (InputAnchor input : getAllInputs()) {
+            input.setVisible(getInputIndex(input) < getBowtieIndex());
+            if(input.isConnected()){
```

Comment:
Hoe heb jij zo snel de bowtie index verandert terwijl het programma draaide?

Daarnaast heb ik nu gefixed wat je zei, los van het feit dat een connection nu nog wel rond blijft zweven, hiervoor is Connection.remove() nodig die in PR 82 zit / komt.